opensearch_py_ml/ml_models/sentencetransformermodel.py

```diff
@@ -766,7 +791,74 @@ def save_as_pt(
                 arcname="tokenizer.json",
             )
         print("zip file is saved to ", zip_file_path, "\n")
-        return compiled_model
+
+    def save_as_onnx(
+        self,
+        model_id="sentence-transformers/msmarco-distilbert-base-tas-b",
+        model_name: str = None,
+        output_path: str = None,
+        zip_file_name: str = None,
+    ) -> None:
+        """
+        download sentence transformer model directly from huggingface, convert model to onnx format,
+        zip the model file and its tokenizer.json file to prepare to upload to the Open Search cluster
+
+        :param model_id:
+            sentence transformer model id to download model from sentence transformers.
+            default model_id = "sentence-transformers/msmarco-distilbert-base-tas-b"
+        :type model_id: string
+        :param model_name:
+            Optional, model name to name the model file, e.g, "sample_model.pt". If None, default takes the
+            model_id and add the extension with ".pt"
+        :type model_name: string
+        :param output_path:
+             Optional, path to save model json file, e.g, "home/save_pre_trained_model_json/"). If None, default as
+             default_folder_path from the constructor
+        :type output_path: string
+        :param zip_file_name:
+            Optional, file name for zip file. e.g, "sample_model.zip". If None, default takes the model_id
+            and add the extension with ".zip"
+        :type zip_file_name: string
+        :return: None. Model file saves in a zip with the tokenzier.json file in local file system
+        :rtype: None
+        """
+
+        model = SentenceTransformer(model_id)
+
+        if model_name is None:
+            model_name = str(model_id.split("/")[-1] + ".onnx")
+
+        model_path = os.path.join(self.folder_path, "onnx", model_name)
+
+        if output_path is None:
+            output_path = self.folder_path
+
+        if zip_file_name is None:
+            zip_file_name = str(model_id.split("/")[-1] + ".zip")
+
+        zip_file_path = os.path.join(self.folder_path, zip_file_name)
+
+        # save tokenizer.json in output_path
+        model.save(output_path)
+
+        convert(
+            framework="pt",
+            model=model_id,
+            output=Path(model_path),
+            opset=16,
+        )
+
+        # zip model file along with tokenizer.json as output
+        with ZipFile(str(zip_file_path), "w") as zipObj:
+            zipObj.write(
+                model_path,
+                arcname=str(model_name),
+            )
+            zipObj.write(
+                os.path.join(output_path, "tokenizer.json"),
+                arcname="tokenizer.json",
+            )
+        print("zip file is saved to ", zip_file_path, "\n")
```

Comment:
Should we return the `zip_file_path` ? So user can use it to upload to ml-commons directly.

Something like
```
model_zip_file = save_as_onnx(xx)
upload_model(model_zip_file)
```

Reply:
Yup, that's a good idea.